src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -95,42 +94,42 @@ export class AccessibleBufferWidget extends DisposableStore {
 		this._xterm.raw.focus();
 	}
 
-	async show(): Promise<void> {
-		const commandDetection = this._capabilities.get(TerminalCapability.CommandDetection);
-		const fragment = !!commandDetection ? this._getShellIntegrationContent() : this._getAllContent();
-		const model = await this._getTextModel(URI.from({ scheme: AccessibleBufferConstants.Scheme, fragment }));
-		if (model) {
-			this._bufferEditor.setModel(model);
+	private async _updateContent(): Promise<ITextModel> {
+		const model = await this._getTextModel(URI.from({ scheme: AccessibleBufferConstants.Scheme, fragment: this._getContent() }));
+		if (!model) {
+			throw new Error('Could not create accessible buffer editor model');
 		}
+		this._bufferEditor.setModel(model);
```

Comment:
We probably want to use pushEditOperations here:

https://github.com/microsoft/vscode/blob/7b39d0e24ae473beb9d1d28d8372e960681ddda9/src/vs/editor/common/model.ts#L1042-L1050

If you're replacing everything, that would need to re-construct the entire backing data structure every time an update occurs (which could be very frequently)

Reply:
Yeah, that looks like what I was trying to find yesterday.